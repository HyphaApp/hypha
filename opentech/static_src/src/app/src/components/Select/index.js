import React from 'react'
import PropTypes from 'prop-types'

const Option = ({ value, display }) => {
    return <option value={value}>{display || value}</option>
}

Option.propTypes = {
    value: PropTypes.string.isRequired,
    display: PropTypes.string,
}

const Select = ({ options, onChange }) => {
    const items = [{ display: '---', value: '' }].concat(options)
    return (
        <select onChange={evt => onChange(evt.target.value)} >
            {items.map(({ value, display }) =>
                <Option key={value} value={value} display={display} />
            )}
        </select>
    )
}


Select.propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
        value: PropTypes.string.isRequired,
        display: PropTypes.string,
    })),
    onChange: PropTypes.func,
}


export default Select
