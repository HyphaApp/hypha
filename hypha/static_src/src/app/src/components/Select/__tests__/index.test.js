import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import Select from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });
import { Option } from '../index'

describe("Test select component", () => {
    const options = [{display : "display1", value : "value1"}]
    const onChange = jest.fn()

    const subject = mount(<Select
      options={options}
      onChange={onChange}
      
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('select').length).toBe(1);
      expect(subject.find('option').first().text()).toBe("---");
      subject.find('select').props().onChange({target :{ value : 1 }})
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith(1);
      expect(subject.containsMatchingElement(<Option />)).toEqual(true);
    });
  
    test('render a select component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test Option in select component", () => {
    const value = "value1"
    const display = "display"

    const subject = mount(<Option
      value={value}
      display={display}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('option').text()).toBe(display);
      expect(subject.find('option').props().value).toBe(value);
    });
  
    test('render a Option in select component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

