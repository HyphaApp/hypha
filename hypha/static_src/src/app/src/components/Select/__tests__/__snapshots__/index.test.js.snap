// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Option in select component render a Option in select component 1`] = `
<Option
  display="display"
  value="value1"
>
  <option
    value="value1"
  >
    display
  </option>
</Option>
`;

exports[`Test select component render a select component 1`] = `
<Select
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  options={
    Array [
      Object {
        "display": "display1",
        "value": "value1",
      },
    ]
  }
>
  <select
    onChange={[Function]}
  >
    <Option
      display="---"
      key=""
      value=""
    >
      <option
        value=""
      >
        ---
      </option>
    </Option>
    <Option
      display="display1"
      key="value1"
      value="value1"
    >
      <option
        value="value1"
      >
        display1
      </option>
    </Option>
  </select>
</Select>
`;
